File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/TVClient.java
package uk.co.ourfriendirony.medianotifier.clients;

import java.io.IOException;
import java.util.List;

import uk.co.ourfriendirony.medianotifier.clients.tmdb.TMDBClient;
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem;

public class TVClient implements Client {

    @Override
    public List<MediaItem> searchMediaItem(String name) throws IOException {
        return new TMDBClient().queryTVShow(name);
    }

    @Override
    public MediaItem getMediaItem(String id) throws IOException {
        return new TMDBClient().getTVShow(id);
    }
}

File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/TVClient.kt
package uk.co.ourfriendirony.medianotifier.clients

import uk.co.ourfriendirony.medianotifier.clients.tmdb.TMDBClient
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem
import java.io.IOException

class TVClient : Client {
    @Throws(IOException::class)
    override fun searchMediaItem(name: String?): List<MediaItem?> {
        return TMDBClient().queryTVShow(name)
    }

    @Throws(IOException::class)
    override fun getMediaItem(id: String?): MediaItem {
        return TMDBClient().getTVShow(id)
    }
}